src/sinks/influxdb/logs.rs

```diff
@@ -0,0 +1,657 @@
+use crate::event::Value;
+use crate::sinks::influxdb::{
+    encode_namespace, encode_timestamp, healthcheck, influx_line_protocol, influxdb_settings,
+    Field, InfluxDB1Settings, InfluxDB2Settings,
+};
+use crate::sinks::util::encoding::EncodingConfigWithDefault;
+use crate::sinks::util::http::{BatchedHttpSink, HttpSink};
+use crate::sinks::util::{BatchBytesConfig, Buffer, TowerRequestConfig};
+use crate::{
+    event::{log_schema, Event},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use futures01::Sink;
+use http::{Method, Request};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap};
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct InfluxDBLogsConfig {
+    pub namespace: String,
+    pub endpoint: String,
+    #[serde(default)]
+    pub tags: Vec<String>,
+    #[serde(flatten)]
+    pub influxdb1_settings: Option<InfluxDB1Settings>,
+    #[serde(flatten)]
+    pub influxdb2_settings: Option<InfluxDB2Settings>,
+    #[serde(
+        skip_serializing_if = "crate::serde::skip_serializing_if_default",
+        default
+    )]
+    pub encoding: EncodingConfigWithDefault<Encoding>,
+    #[serde(default)]
+    pub batch: BatchBytesConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        retry_attempts: Some(5),
+        ..Default::default()
+    };
+}
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+#[derivative(Default)]
+pub enum Encoding {
+    #[derivative(Default)]
+    Default,
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<InfluxDBLogsConfig>("influxdb_logs")
+}
+
+#[typetag::serde(name = "influxdb_logs")]
+impl SinkConfig for InfluxDBLogsConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let healthcheck = healthcheck(
+            self.clone().endpoint,
+            self.clone().influxdb1_settings,
+            self.clone().influxdb2_settings,
+            cx.resolver(),
+        )?;
```

Comment:
This looks wasteful, with `self` being cloned repeatedly. Better might be to move the `healthcheck` function inside `SinkConfig` which would reduce this down to: `let healthcheck = self.healthcheck(cx.resolver())?;`

Reply:
done